dojson/overdo.py

```diff
@@ -62,25 +62,29 @@ def _query(self, key):
 
     def do(self, blob):
         """Translate blob values and instantiate new model instance."""
-        output = {}
+        output = []
 
         if self.index is None:
             self.build()
 
-        for key, value in iteritems(blob):
+        if isinstance(blob, GroupableOrderedDict):
+            items = blob.iteritems(repeated=True)
+        else:
+            items = iteritems(blob)
+
+        for key, value in items:
             for name, creator, field in self._query(key):
                 try:
                     data = creator(output, key, value)
-                    if getattr(creator, '__extend__', False):
```

Comment:
duck typing instead of isinstance. The problem lies in the what is done with it.

The XML serialization of a repeatable field is equivalent to the one of a non-repeatable field. I don't get why it should be different in the JSON serialization of the same data.